packages/x-date-pickers/src/tests/describeGregorianAdapter/testCalculations.ts

```diff
@@ -1,275 +1,529 @@
+        ]),
+      ).to.equal(true);
+
+      expect(
+        adapter.isWithinRange(adapter.date('2019-12-01T00:00:00.000Z')!, [
+          adapter.date('2019-09-01T00:00:00.000Z')!,
+          adapter.date('2019-11-01T00:00:00.000Z')!,
+        ]),
+      ).to.equal(false);
+
+      expect(
+        adapter.isWithinRange(adapter.date('2019-10-01')!, [
+          adapter.date('2019-09-01')!,
+          adapter.date('2019-11-01')!,
+        ]),
+      ).to.equal(true);
+
+      expect(
+        adapter.isWithinRange(adapter.date('2019-12-01')!, [
+          adapter.date('2019-09-01')!,
+          adapter.date('2019-11-01')!,
+        ]),
+      ).to.equal(false);
+    });
+
+    it('should use inclusivity of range', () => {
+      expect(
+        adapter.isWithinRange(adapter.date('2019-09-01T00:00:00.000Z')!, [
+          adapter.date('2019-09-01T00:00:00.000Z')!,
+          adapter.date('2019-12-01T00:00:00.000Z')!,
+        ]),
+      ).to.equal(true);
+
+      expect(
+        adapter.isWithinRange(adapter.date('2019-12-01T00:00:00.000Z')!, [
+          adapter.date('2019-09-01T00:00:00.000Z')!,
+          adapter.date('2019-12-01T00:00:00.000Z')!,
+        ]),
+      ).to.equal(true);
+
+      expect(
+        adapter.isWithinRange(adapter.date('2019-09-01')!, [
+          adapter.date('2019-09-01')!,
+          adapter.date('2019-12-01')!,
+        ]),
+      ).to.equal(true);
+
+      expect(
+        adapter.isWithinRange(adapter.date('2019-12-01')!, [
+          adapter.date('2019-09-01')!,
+          adapter.date('2019-12-01')!,
+        ]),
+      ).to.equal(true);
+    });
   });
 
   it('Method: startOfYear', () => {
-    expect(adapter.formatByString(adapter.startOfYear(testDate), formatDateTime)).to.equal(
-      '2018-01-01 00:00:00',
-    );
+    const expected = '2018-01-01T00:00:00.000Z';
+    expect(adapter.startOfYear(testDateIso)).toEqualDateTime(expected);
+    expect(adapter.startOfYear(testDateLocale)).toEqualDateTime(expected);
   });
 
   it('Method: startOfMonth', () => {
-    expect(adapter.formatByString(adapter.startOfMonth(testDate), formatDateTime)).to.equal(
-      '2018-10-01 00:00:00',
-    );
+    const expected = '2018-10-01T00:00:00.000Z';
+    expect(adapter.startOfMonth(testDateIso)).toEqualDateTime(expected);
+    expect(adapter.startOfMonth(testDateLocale)).toEqualDateTime(expected);
   });
 
   it('Method: startOfWeek', () => {
-    expect(adapter.formatByString(adapter.startOfWeek(testDate), formatDateTime)).to.equal(
-      adapter.lib === 'luxon' ? '2018-10-29 00:00:00' : '2018-10-28 00:00:00',
-    );
-
-    // Non ISO
-    expect(
-      adapter.formatByString(
-        adapter.startOfWeek(adapter.date('2018-10-28T00:00:00.000Z')!),
-        formatDateTime,
-      ),
-    ).to.equal(adapter.lib === 'luxon' ? '2018-10-22 00:00:00' : '2018-10-28 00:00:00');
+    const expected =
+      adapter.lib === 'luxon' ? '2018-10-29T00:00:00.000Z' : '2018-10-28T00:00:00.000Z';
```

Comment:
Did you consider changing the `adapter` locale in tests to one where weeks start on Monday in order to avoid the need for these checks? 🤔

Reply:
I like to keep the default locale because it's more broadly known.
But I agree that those exceptions are not great